src/sentry/profiles/task.py

```diff
@@ -136,6 +137,24 @@ def process_profile_task(
         set_measurement("profile.stacks.processed", len(profile["profile"]["stacks"]))
         set_measurement("profile.frames.processed", len(profile["profile"]["frames"]))
 
+    if (
+        enabled := options.get("profiling.generic_metrics.functions_ingestion.enabled")
+    ) and project.organization_id in (
+        allowed_orgs := options.get("profiling.generic_metrics.functions_ingestion.allowed_org_ids")
+    ):
+        try:
+            project_key, _ = ProjectKey.objects.get_or_create(
```

Comment:
We might want to call `get_from_cache` first to avoid going to the db on every profile, like we do here:

https://github.com/getsentry/sentry/blob/1c5a430b55fb265102c1cf5687f9de1dbed6bcf4/src/sentry/event_manager.py#L507

Reply:
as we discussed offline, even if we added `project` and `use_case` as cachable keys, `get_from_cache` only allows to specify one field by which we can cache a value. See --> https://github.com/getsentry/sentry/blob/master/src/sentry/db/models/manager/base.py#L312

I still think some caching is a sensible thing here, but for the reasons mentioned above, I'll have to think about an alternative solution.